changelogs/unreleased/add-asan-flag-to-tarantool-build.md

```diff
@@ -1,4 +0,0 @@
-## feature/box
-
-* Added a new flag `tarantool.build.asan` that shows whether build
-  flag `ENABLE_ASAN` is set.
```

Comment:
I agree that it is a small feature. However, it is a feature: here we add something into the public API.

AFAIR, we need it for testing purposes. The obvious workaround would be to move the flag to some internal API, however it would complicate backporting of new test cases.

Can we drop the flag from the executable, but add a workaround into test/helper.lua? This file is loaded at luatest startup and can be used to patch the same `tarantool.build` table.

OTOH, I understand that the feature is small and safe, so we can also consider some alternatives toward the policy change. I would note that we need some separate release notes category for such changes. 'Functionality added or changed' shouldn't appear in a bugfix release.

Let's discuss?

Reply:
I would question it is a feature. Reading the documentation:

> one can answer some questions about how the tarantool server was built, such as “what flags were used”, or “what was the version of the compiler”

There's no promises regarding fields in has or has not, neither any reasonable "public" inference from the facts it contains. E.g. what should you - as a client - take from the cmake options used while tarantool was built? Will you make a choice in favor of '-Wpedantic'? It says "some questions" so it has no limitation in my opinion. 

Totally different story are `pid()` and `uptime()`, which are not only pinned by the doc, but advised for use:

> Additionally one can see the uptime and the server version and the process id. Those information items can also be accessed with [box.info()](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_info/#box-introspection-box-info) but use of the tarantool module is recommended.

So for this particular case I believe it doesn't represent a feature that we have to find a way to introduce in a patch revision. Not saying it can't be a real problem per se: sometimes we need some feature to be backported. 
I propose to introduce a **'waiver'** (in terms of our policy) status for such cases and discuss it every time it appears. It will designate a feature/change in functionality we intentionally introduce with understanding we violate the policy (waiving it).

Reply:
Let's move it to its own category `testing`. We need to add some words that this is to ease backporting of tests from master.

Reply:
Sure, will do after categories backport.

Reply:
I'll left the comment floating here open till the update.

Reply:
done, have a look in https://github.com/tarantool/tarantool/pull/9448/commits/eb0c06411ab507f83a6dae6d9a5ec16026610d2f